Update Readme (#34)
DevExpress-Examples/devextreme-datagrid-select-all-checkboxes: README.md

[![](https://img.shields.io/badge/💬_Leave_Feedback-feecdd?style=flat-square)](#does-this-example-address-your-development-requirementsobjectives)
<!-- default badges end -->

> # DevExtreme DataGrid - Implement Three-State "Select All" Checkboxes in Group Rows

Reviewer: ```suggestion
# DevExtreme DataGrid - Implement Three-State "Select All" Checkboxes Within Group Rows
```

> This example implements custom "Select All" checkboxes in DataGrid group rows using the [groupCellTemplate](https://js.devexpress.com/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#groupCellTemplate) API. These checkboxes support three states: checked, unchecked, and indeterminate (displayed when some but not all rows are selected in a group).

Reviewer: I keep in mind that Ray asked us not to overuse "implements". I also shortened a bit this paragraph.

```suggestion
This example uses a [groupCellTemplate](https://js.devexpress.com/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#groupCellTemplate] to add custom "Select All" checkboxes to DataGrid group rows. These checkboxes can have one of the following states: checked, unchecked, and indeterminate (displayed when multiple rows are selected within a group).
```

> ![DevExtreme DataGrid - Implement Three-State "Select All" Checkboxes in Group Rows](images/devextreme-datagrid-select-all-checkbox.png)

Reviewer: I think we can benefit from a gif here.

> This example binds the DataGrid to a remote data source and enables [remote operations](https://js.devexpress.com/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/remoteOperations/).

Reviewer: This sentence looks out of place here. Let's move it to the intro or to the explanation section.

> To determine group row checkbox states and select/unselect rows, this example uses the [selectedRowKeys](https://js.devexpress.com/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/#selectedRowKeys) value. To implement this functionality in a DataGrid where [deferred selection](https://js.devexpress.com/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/selection/#deferred) is enabled, use [selectionFilter](https://js.devexpress.com/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/#selectionFilter) instead. Refer to the following help topic for additional information about deferred selection: [DevExtreme DataGrid - Deferred Selection](https://js.devexpress.com/Documentation/Guide/UI_Components/DataGrid/Enhance_Performance_on_Large_Datasets/#Deferred_Selection).

Reviewer: We can make a list out of this paragraph. This will save us a lot of words :)

[![](https://img.shields.io/badge/💬_Leave_Feedback-feecdd?style=flat-square)](#does-this-example-address-your-development-requirementsobjectives)
<!-- default badges end -->

# DevExtreme DataGrid - Implement Three-State "Select All" Checkboxes Within Group Rows

> This example uses a [groupCellTemplate](https://js.devexpress.com/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#groupCellTemplate) to add custom "Select All" checkboxes to DataGrid group rows. These checkboxes can have one of the following states: checked, unchecked, and indeterminate (displayed when multiple rows are selected within a group). This example binds the DataGrid to a remote data source and enables [remote operations](https://js.devexpress.com/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/remoteOperations/).

Reviewer: ```suggestion
This example uses a [groupCellTemplate](https://js.devexpress.com/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#groupCellTemplate) to add custom "Select All" checkboxes to DataGrid group rows. These checkboxes can have one of the following states: checked, unchecked, and indeterminate (when multiple rows are selected within a group). This example binds the DataGrid to a remote data source and enables [remote operations](https://js.devexpress.com/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/remoteOperations/).
```

![DevExtreme DataGrid - Implement Three-State "Select All" Checkboxes in Group Rows](images/devextreme-datagrid-select-all-checkbox.png)

> To determine group row checkbox states and select/unselect rows, use the following properties:

Reviewer: ```suggestion
To determine group row checkbox states and select/deselect rows, use the following properties:
```

- [selectionFilter](https://js.devexpress.com/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/#selectionFilter): If [selection.deferred](https://js.devexpress.com/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/selection/#deferred) is enabled

> - [selectedRowKeys](https://js.devexpress.com/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/#selectedRowKeys): If deferred selection is not enabled

Reviewer: ```suggestion
- [selectionFilter](https://js.devexpress.com/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/#selectionFilter) (if [selection.deferred](https://js.devexpress.com/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/selection/#deferred) is enabled)
- [selectedRowKeys](https://js.devexpress.com/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/#selectedRowKeys) (in other cases)
```

    - [Home.vue](Vue/src/components/HomeContent.vue)
    - [GroupRowComponent.vue](Vue/src/components/GroupRowSelection/GroupRowComponent.vue)
    - [GroupRowSelectionHelper.ts](Vue/src/components/GroupRowSelection/GroupRowSelectionHelper.ts)
      

> ## Implementation details

Reviewer: I suggest that we keep the "Implementation Details" heading **if** we can add a code snippet below.

Author: The implementations across frameworks vary by quite a lot in this example, so there's no shared, short bit of code we could add here.
